applications/FSIapplication/python_scripts/fsi_analysis.py

```diff
@@ -0,0 +1,338 @@
+
+        # Deprecation warnings
+        # This makes possible the FSI solver derivation from the core base python_solver.py
+        if not self.project_parameters.Has("echo_level"):
+            Kratos.Logger.PrintInfo("FSIAnalysis", "Using the old way to pass the echo_level, this will be removed!")
+            self.project_parameters.AddEmptyValue("echo_level")
+            self.project_parameters["echo_level"].SetInt(self.echo_level)
+
+        # Add solver variables (note that the solver is created in the first _GetSolver() call)
+        self._GetSolver().AddVariables()
+
+    def Initialize(self):
+        '''
+        Construct and initialize all classes and tools used in the simulation loop.
+        '''
+        self._SetUpRestart()
+
+        if self.load_restart:
+            fluid_restart_utility = self._GetFluidRestartUtility()
+            structure_restart_utility = self._GetStructureRestartUtility()
+            fluid_restart_utility.LoadRestart()
+            structure_restart_utility.LoadRestart()
+        else:
+            self._GetSolver().ImportModelPart()
+            self._GetSolver().PrepareModelPart()
+            self._GetSolver().AddDofs()
+            self.fluid_main_model_part = self.model.GetModelPart(
+                self.project_parameters["fluid_solver_settings"]["solver_settings"]["model_part_name"].GetString())
+            self.structure_main_model_part = self.model.GetModelPart(
+                self.project_parameters["structure_solver_settings"]["solver_settings"]["model_part_name"].GetString())
+
+        # This should let eventual derived stages modify the model after reading.
+        self.ModifyInitialProperties()
+        self.ModifyInitialGeometry()
+
+        self._SetUpListOfProcesses()
+        self._SetUpAnalysis()
+
+        for process in self._GetListOfProcesses():
+            process.ExecuteBeforeSolutionLoop()
+
+    def InitializeSolutionStep(self):
+
+        # Since no substepping is implemented yet, structure and fluid step/time must match
+        step_fluid = self.fluid_main_model_part.ProcessInfo[KratosMultiphysics.STEP]
+        step_structure = self.structure_main_model_part.ProcessInfo[KratosMultiphysics.STEP]
+        if step_fluid != step_structure:
+            err_msg =  'Fluid step: '+ str(step_fluid) + '\n'
+            err_msg += 'Structure step: '+ str(step_structure) + '\n'
+            err_msg += 'No substepping has been implemented yet. Fluid and structure step must match.'
+            raise Exception(err_msg)
+
+        if self.is_printing_rank:
+            Kratos.Logger.PrintInfo(self._GetSimulationName(),"STEP = ", step_fluid)
+            Kratos.Logger.PrintInfo(self._GetSimulationName(),"TIME = ", self.time)
+
+        self.ApplyBoundaryConditions() #here the processes are called
+        self.ChangeMaterialProperties() #this is normally empty
+        self._GetSolver().InitializeSolutionStep()
+
+    def OutputSolutionStep(self):
```

Comment:
If you use the new way for the output (and the restart) then you don't need this function
=> have a look at the structure or the fluid stage